src/hcs-2/browser.ts

```diff
@@ -0,0 +1,383 @@
+      
+      return {
+        success: true,
+        receipt,
+        sequenceNumber: receipt.topicSequenceNumber?.low ?? undefined
+      };
+    } catch (error) {
+      this.logger.error(`Failed to migrate registry: ${error}`);
+      return {
+        success: false,
+        error: `Failed to migrate registry: ${error}`
+      };
+    }
+  }
+
+  /**
+   * Get all entries from a registry
+   * @param topicId The topic ID of the registry
+   * @param options Query options
+   * @returns Promise resolving to the registry information
+   */
+  async getRegistry(topicId: string, options: QueryRegistryOptions = {}): Promise<TopicRegistry> {
+    try {
+      // Get topic info to determine registry type
+      const topicInfo = await this.getTopicInfo(topicId);
+      const memoInfo = this.parseRegistryTypeFromMemo(topicInfo.memo);
+      
+      if (!memoInfo) {
+        throw new Error(`Topic ${topicId} is not an HCS-2 registry (invalid memo format)`);
+      }
+      
+      // Get messages from the topic
+      const messages = await this.mirrorNode.getTopicMessages(
+        topicId,
+        {
+          limit: options.limit ?? 100,
+          order: options.order ?? 'asc'
+        }
+      );
+      
+      // Parse messages into registry entries
+      return this.parseRegistryEntries(
+        topicId,
+        messages,
+        memoInfo.registryType,
+        memoInfo.ttl
+      );
+    } catch (error) {
+      this.logger.error(`Failed to get registry: ${error}`);
+      throw error;
+    }
+  }
+  
+  /**
+   * Submit a message to a topic
+   * @param topicId The topic ID to submit to
+   * @param payload The message payload
+   * @returns Promise resolving to the transaction receipt
+   */
+  async submitMessage(topicId: string, payload: HCS2Message): Promise<TransactionReceipt> {
+    try {
+      // Validate message
+      const { valid, errors } = this.validateMessage(payload);
+      if (!valid) {
+        throw new Error(`Invalid HCS-2 message: ${errors.join(', ')}`);
+      }
+      
+      // Create transaction
+      const transaction = new TopicMessageSubmitTransaction()
+        .setTopicId(TopicId.fromString(topicId))
+        .setMessage(JSON.stringify(payload));
+      
+      // Execute transaction through WalletConnect - use any to avoid type issues
+      const txResponse = await (this.hwc as any).executeTransactionWithErrorHandling(
+        transaction,
+        false,
+      );
+      
+      if (txResponse?.error) {
+        throw new Error(txResponse.error);
+      }
+      
+      return txResponse.result;
+    } catch (error) {
+      this.logger.error(`Failed to submit message: ${error}`);
+      throw error;
+    }
+  }
+
+  /**
+   * Get information about a topic
+   * @param topicId The topic ID
+   * @returns Promise resolving to the topic information
+   */
+  private async getTopicInfo(topicId: string): Promise<any> {
```

Comment:
since this just calls the mirrorNode client it doesn't make sense to wrap the method